agent/grpc-external/services/resource/write.go

```diff
@@ -142,10 +151,11 @@ func (s *Server) Write(ctx context.Context, req *pbresource.WriteRequest) (*pbre
 		return nil, status.Error(codes.Aborted, err.Error())
 	case errors.Is(err, storage.ErrWrongUid):
 		return nil, status.Error(codes.FailedPrecondition, err.Error())
+	case isGRPCStatusError(err):
+		return nil, err
```

Comment:
Not a big fan of this for passing through already constructed GRPC status errors. Time to introduce private resource service level errors and do conversion in this switch maybe?